src/Paket/Program.fs

```
@@ -301,6 +301,76 @@ let push (results : ParseResults<_>) =
+                | Paket.Requirements.FrameworkRestrictionList list ->
+                  list |> Seq.collect (
+                    function
+                    | Paket.Requirements.FrameworkRestriction.Exactly framework
+                    | Paket.Requirements.FrameworkRestriction.AtLeast framework -> Seq.singleton framework
+                    | Paket.Requirements.FrameworkRestriction.Between (bottom,top) -> [bottom; top] |> Seq.ofList //TODO: do we need to cap the list of generated frameworks based on this? also see todo in Requirements.fs for potential generation of range for 'between'
+                    | Paket.Requirements.FrameworkRestriction.Portable portable -> failwithf "unhandled portable framework %s" portable
+                  )
+              )
+            |> Seq.concat
+    )
+
+    let environmentFramework = lazy (
+        // HACK: resolve .net version based on environment
+        // list of match is incomplete / inaccurate
+        let version = Environment.Version
+        match version.Major, version.Minor, version.Build, version.Revision with
+        | 4, 0, 30319, 42000 -> DotNetFramework (FrameworkVersion.V4_6)
+        | 4, 0, 30319, _ -> DotNetFramework (FrameworkVersion.V4_5)
+        | _ -> DotNetFramework (FrameworkVersion.V4_5) // paket.exe is compiled for framework 4.5
+    )
+
+    let frameworksToGenerate =
+        let targetFrameworkList = providedFrameworks |> List.choose FrameworkDetection.Extract
+        
+        if targetFrameworkList.Length <> providedFrameworks.Length then
+            // print out bogus frameworks
+            providedFrameworks
+            |> Seq.map (fun f -> f, FrameworkDetection.Extract f)
+            |> Seq.filter (snd >> Option.isNone)
+            |> Seq.iter (fst >> (traceErrorfn "framework %s was unrecognized"))
+            // not sure how we should exit instead of generating for default framework
```

IMHO we should throw an exception if one of the parameters was not recognized (fail fast & fail early).